pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -31,14 +31,33 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/config"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+	kccpredicate "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/predicate"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
 	"github.com/googleapis/gax-go/v2"
 )
 
 const ctrlName = "sqlinstance-controller"
 
 func init() {
-	registry.RegisterModel(krm.SQLInstanceGVK, newSQLInstanceModel)
+	rolloutGate := &SQLInstanceRolloutGate{}
+	registry.RegisterModelWithRolloutGate(krm.SQLInstanceGVK, newSQLInstanceModel, rolloutGate)
```

Comment:
Tip: This is fine, another pattern is to create a struct to give something like Python named arguments e.g.

```
registry.RegisterModel(registry.Options{GVK: krm.SQLInstanceGVK, ModelFn: newSQLInstanceModel, RolloutGate: rolloutGate}
```

At 3 arguments, not a problem.  But when the number creeps up, particularly if it's a variable number of arguments, becomes more useful.

Reply:
I kinda intentionally created another function for this (and didn't change the original function signature), in order to sort of "hide" this feature. Not really sure that we want to encourage direct reconciler developers to use this capability unless there's a strong need. Thought it might be desirable to have the default / most obvious way to be without the predicate / gate, instead (for most resources) we would want to just switch over to direct reconciler completely.